gammapy/spectrum/dataset.py

```diff
@@ -195,6 +195,85 @@ def __init__(
         self.edisp = edisp
         self.model = model
 
+    @property
+    def livetime(self):
+        return self._livetime
+
+    @livetime.setter
+    def livetime(self, livetime):
+        self._livetime = livetime
+        if self.counts_on is not None:
+            self.counts_on.livetime = livetime
+        # TODO : check if off might have different exposure
+        if self.counts_off is not None:
+            self.counts_off.livetime = livetime
+
+    @property
+    def lo_threshold(self):
+        """Low energy threshold"""
+        return self.counts_on.lo_threshold
+
+    @lo_threshold.setter
+    def lo_threshold(self, threshold):
+        self.counts_on.lo_threshold = threshold
+        if self.counts_off is not None:
+            self.counts_off.lo_threshold = threshold
+
+    @property
+    def hi_threshold(self):
+        """High energy threshold"""
+        return self.counts_on.hi_threshold
+
+    @hi_threshold.setter
+    def hi_threshold(self, threshold):
+        self.counts_on.hi_threshold = threshold
+        if self.counts_off is not None:
+            self.counts_off.hi_threshold = threshold
+
+    def reset_thresholds(self):
```

Comment:
Is this method really needed? Maybe we can just allow `dataset.hi_threshold = None` and `dataset.lo_threshold`? This would be in line with what we have in the `set_fit_energy_range()` method and `geom.energy_mask()`.